### PR TITLE
[OBSDEF-46383] Show pagination options if at least 1 record is present

### PR DESCRIPTION
**Pull request by netraja-chavan**
**JIRA:-**
https://jira.cec.lab.emc.com/browse/OBSDEF-46383

Before:-
image

After:-
image

Test link:-
Storybook - http://10.227.244.41:6006/?path=/story/datagrid--grid-with-pagination-and-default-pagesizes-dropdown

### CODE REVIEW

File: src/datagrid/DataGrid.tsx [hunk 1]
```diff
@@ -640,15 +640,23 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
...
+            //If page size selected by user is greater than maximum limit for custom page size
+            else if (pageSize > maxCustomPageSize && this.customPageSizeRef.current) {
+                //If total number of records is less than maximum limit for custom page size
+                if (totalItems < maxCustomPageSize) {
+                    this.customPageSizeRef.current.value = totalItems.toString();
+                    this.getPage(DEFAULT_CURRENT_PAGE_NUMBER, parseInt(this.customPageSizeRef.current.value));
+                } else {
+                    this.customPageSizeRef.current.value = maxCustomPageSize.toString();
+                    this.getPage(DEFAULT_CURRENT_PAGE_NUMBER, parseInt(this.customPageSizeRef.current.value));
+                }
```
Comment (akate1) [inline src/datagrid/DataGrid.tsx:659]:
```suggestion
                           this.customPageSizeRef.current.value =  (totalItems < maxCustomPageSize) ?  totalItems.toString() : maxCustomPageSize.toString();
                             this.getPage(DEFAULT_CURRENT_PAGE_NUMBER, parseInt(this.customPageSizeRef.current.value));
                
```

**Review by akate1 — APPROVED** [review]
Tested
Left 1 small comment please take care of it 
LGTM

File: src/datagrid/DataGrid.tsx (hunk 1 above)
Comment (netraja-chavan) [reply src/datagrid/DataGrid.tsx]: done

**Review by GauravMenezes-Professional — APPROVED** [review]
Tested storybook link with custom value, pagination occurs properly. LGTM

**Review by psivakumaran — APPROVED** [review]
LGTM

**Pull request merged**